managed/models/encryption_helpers.go

```diff
@@ -0,0 +1,336 @@
+// Copyright (C) 2023 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package models
+
+import (
+	"database/sql"
+	"encoding/json"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/percona/pmm/managed/utils/encryption"
+)
+
+// EncryptAgent encrypt agent.
+func EncryptAgent(agent *Agent) {
+	agentEncryption(agent, encryption.Encrypt)
+}
+
+// DecryptAgent decrypt agent.
+func DecryptAgent(agent *Agent) {
+	agentEncryption(agent, encryption.Decrypt)
+}
+
+func agentEncryption(agent *Agent, handler func(string) (string, error)) {
+	if agent.Username != nil {
+		username, err := handler(*agent.Username)
+		if err != nil {
+			logrus.Debugf("username:%s", username)
+			logrus.Warning(err)
+		}
+		agent.Username = &username
+	}
+
+	if agent.Password != nil {
+		password, err := handler(*agent.Password)
+		if err != nil {
+			logrus.Debugf("password:%s", password)
+			logrus.Warning(err)
+		}
+		agent.Password = &password
+	}
+
+	if agent.AgentPassword != nil {
+		agentPassword, err := handler(*agent.AgentPassword)
+		if err != nil {
+			logrus.Debugf("agent_password:%s", agentPassword)
+			logrus.Warning(err)
+		}
+		agent.AgentPassword = &agentPassword
+	}
+
+	if agent.AWSAccessKey != nil {
+		awsAccessKey, err := handler(*agent.AWSAccessKey)
+		if err != nil {
+			logrus.Debugf("aws_access_key:%s", awsAccessKey)
+			logrus.Warning(err)
+		}
+		agent.AWSAccessKey = &awsAccessKey
+	}
+
+	if agent.AWSSecretKey != nil {
+		awsSecretKey, err := handler(*agent.AWSSecretKey)
+		if err != nil {
+			logrus.Debugf("aws_secret_key:%s", awsSecretKey)
+			logrus.Warning(err)
+		}
+		agent.AWSSecretKey = &awsSecretKey
+	}
+
+	var err error
+	if agent.MySQLOptions != nil {
```

Comment:
They are used to handle read from specific column in DB (which is not string). For example PostgreSQLOptions column is json. Here it is used to handle agent. So they are working differently.

Reply:
Methods DecryptMySQLOptionsHandler, DecryptPostgreSQLOptionsHandler, DecryptMongoDBOptionsHandler, DecryptAzureOptionsHandler are not used now (since during migration we using only Encrypt ones). I can remove them if you want. But since they are 3 lines of code I originally defines them due to consistency.